### PR TITLE
🏃 upgrade to latest CAPI version and upgrade/cleanup some other deps

### PR DESCRIPTION
**Pull request by sbueringer**
**What this PR does / why we need it**:

Upgrades our dependencies to the latest versions used in CAPI. I also cleaned up hack/tools/go.mod a bit.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [x] squashed commits
- if necessary:
  - [x] includes documentation
  - [x] adds unit tests

/hold

### CODE REVIEW

File: test/e2e/data/e2e_conf.yaml [hunk 1]
```diff
@@ -12,11 +12,11 @@
 managementClusterName: capo-e2e
 
 images:
-- name: gcr.io/k8s-staging-cluster-api/cluster-api-controller:nightly_master_20210310
+- name: gcr.io/k8s-staging-cluster-api/cluster-api-controller:nightly_master_20210402
```
Comment (jichenjc) [inline test/e2e/data/e2e_conf.yaml:15]: one minor question: 

do we want to keep update this time to time ?

Comment (sbueringer) [reply test/e2e/data/e2e_conf.yaml:15]: Yes, will need a PR fein time to time until the release. Current estimation is June

Comment (jichenjc) [reply test/e2e/data/e2e_conf.yaml:15]: ok , thanks for the info~

**Comment by sbueringer** [comment]
@hidekazuna @prankul88 ptal, thx :)

**Comment by prankul88** [comment]
Works for me !

/lgtm

**Pull request merged**